Parser/string_parser.c

```diff
@@ -357,9 +357,15 @@ fstring_compile_expr(Parser *p, const char *expr_start, const char *expr_end,
             break;
         }
     }
+    
     if (s == expr_end) {
-        RAISE_SYNTAX_ERROR("f-string: empty expression not allowed");
-        return NULL;
+        if (*expr_end == '!' || *expr_end == ':' || *expr_end == '=') {
+            RAISE_SYNTAX_ERROR("f-string: expression required before '%c'", *expr_end);
+            return NULL;
+        } else {
```

Comment:
I know this is a nit, but could you remove the else clause, and move the next two statements out one level? When reviewing code it's more obvious that all code paths have the "return NULL" in them.